pinot-query-planner/src/test/java/org/apache/pinot/query/testutils/MockRoutingManagerFactory.java

```diff
@@ -63,25 +66,28 @@ public MockRoutingManagerFactory(int... ports) {
     _schemaMap = new HashMap<>();
     _hybridTables = new HashSet<>();
     _serverInstances = new HashMap<>();
+    _nullHandlingMap = new HashMap<>();
     _tableServerSegmentsMap = new HashMap<>();
     for (int port : ports) {
       _serverInstances.put(toHostname(port), getServerInstance(HOST_NAME, port, port, port, port));
     }
   }
 
-  public void registerTable(Schema schema, String tableName) {
+  public void registerTable(Schema schema, String tableName, boolean enableNullHandling) {
```

Comment:
Is the intention to generate null value vector for the table? IMO we should always generate null value vector. We will persist the null value vector only when there are null values.

Reply:
> IMO we should always generate null value vector. We will persist the null value vector only when there are null values.

I have to be clear. The changes in the test engine were copied from @walterddr PR. Anyway, I think it may be interesting to test the behavior when null vector was not created at ingestion time (IndexingConfig.isNullHandlingEnabled was false at ingestion time).

At the same time, I agree that this situation is very simple (if there is no null vector, we assume there is no null) and therefore I would say it would be better to assume in tests that by default tables IndexingConfig.isNullHandlingEnabled is true and allow to optionally disable it.

I disagree. It is useful to test how we behave in case the file doesn't have null vector and we ask for it. I know it is trivial (we just behave as it was not null value